text/0045-add-notion-of-protocol.md

```diff
@@ -0,0 +1,127 @@
+# Summary
+- Define supported protocols for services and outputs in libobs and OBS Studio
+- Outputs for streaming are registered with one or more compatible protocols and codecs
+- Only codecs compatible with the output and the service are shown
+- Audio codec can be chosen if various compatible, codec in simple mode and encoder advanced output mode.
+- Will be extended with RFC 39
+
+# Motivation
+Create a better management of outputs, encoders, and services with an approach focused on protocols.
+
+# Design
+
+## Outputs API
+Only outputs with the `OBS_OUTPUT_SERVICE` flag are concerned.
+
+Adding to `obs_output_info`:
+- `const char *protocols`: protocols separated with a semi-colon (ex: `"RTMP;RTMPS"`), required to register the output. The string used to identify the protocol must be its officialy/widely used acronym, those acronyms are usually uppercased.
+
+Adding to this API these functions:
+- `const char *obs_output_get_protocols(const obs_output_t *output)`: returns protocols supported by the output.
+- `bool obs_is_output_protocol_registered(const char *protocol)`: return true if an output with the protocol is registered.
+- `bool obs_enum_output_service_types(const char *protocol, size_t idx, const char **id)`: enumerate all outputs types compatible with the given protocol.
```

Comment:
Judging by the description and implementation, there is no relevance to services in this function. Perhaps `obs_enum_outputs_with_protocol()` would serve as a better name?

Reply:
Or `obs_enum_output_types_with_protocol()`

Reply:
Changed.